In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

## Read the datasets

In [2]:
food_production = pd.read_csv('datasets/Africa Food Production.csv')
food_supply = pd.read_csv('datasets/Africa Food Supply.csv')

In [3]:
food_production.head()

Country                Item  Year  Value
0  Algeria  Wheat and products  2004   2731
1  Algeria  Wheat and products  2005   2415
2  Algeria  Wheat and products  2006   2688
3  Algeria  Wheat and products  2007   2319
4  Algeria  Wheat and products  2008   1111

In [4]:
food_items = food_production['Item'].unique()
print(sorted(food_items))

['Alcohol, Non-Food', 'Apples and products', 'Aquatic Animals, Others', 'Aquatic Plants', 'Bananas', 'Barley and products', 'Beans', 'Beer', 'Beverages, Alcoholic', 'Beverages, Fermented', 'Bovine Meat', 'Butter, Ghee', 'Cassava and products', 'Cephalopods', 'Cereals, Other', 'Citrus, Other', 'Cloves', 'Cocoa Beans and products', 'Coconut Oil', 'Coconuts - Incl Copra', 'Coffee and products', 'Cottonseed', 'Cottonseed Oil', 'Cream', 'Crustaceans', 'Dates', 'Demersal Fish', 'Eggs', 'Fats, Animals, Raw', 'Fish, Body Oil', 'Fish, Liver Oil', 'Freshwater Fish', 'Fruits, Other', 'Grapefruit and products', 'Grapes and products (excl wine)', 'Groundnut Oil', 'Groundnuts (Shelled Eq)', 'Honey', 'Lemons, Limes and products', 'Maize Germ Oil', 'Maize and products', 'Marine Fish, Other', 'Meat, Other', 'Milk - Excluding Butter', 'Millet and products', 'Molluscs, Other', 'Mutton & Goat Meat', 'Nuts and products', 'Oats', 'Offals, Edible', 'Oilcrops Oil, Other', 'Oilcrops, Other', 'Olive Oil', 'Oliv

In [5]:
food_supply.head()

Country  Year  Value
0  Algeria  2004   2987
1  Algeria  2005   2958
2  Algeria  2006   3047
3  Algeria  2007   3041
4  Algeria  2008   3048

### Question 1
Which country produced the highest amount of oats in 2010?

In [6]:
item = food_production.groupby(['Country','Item','Year'], as_index=False)['Value'].sum()
year = item[item['Year'] == 2010]
oat = year[year['Item'] == 'Oats'].sort_values(by = 'Value', ascending = False)
oat

Country  Item  Year  Value
326         Algeria  Oats  2010     88
6646       Ethiopia  Oats  2010     48
13526       Morocco  Oats  2010     37
18186  South Africa  Oats  2010     34
9676          Kenya  Oats  2010      3
20356       Tunisia  Oats  2010      2
22816      Zimbabwe  Oats  2010      1
8036          Ghana  Oats  2010      0
10126       Lesotho  Oats  2010      0

### Ans: Algeria - 88kt of oats

### Question 2
Which three countries respectively had the lowest food supply between 2004 and
2006 (inclusive)?

In [7]:
supply = food_supply.groupby(['Country','Year'], as_index=False)['Value'].sum()
supply_years = supply[supply['Year']<= 2006]
country = supply_years.groupby('Country')['Value'].sum().sort_values()
country.head(3)

Country
Zambia                      5577
Ethiopia                    5827
Central African Republic    6044
Name: Value, dtype: int64

### Ans: Zambia - 5577, Ethiopia - 5827, and Central African Republic - 6044

### Question 5
What three countries have produced the highest total quantity of rice from 2004 to
2013?

In [8]:
items= food_production.groupby(['Country','Item'], as_index = False)['Value'].sum()
rice = items[items['Item'] == 'Rice (Milled Equivalent)'].sort_values(by = 'Value', ascending = False)
rice.head(3)

Country                      Item  Value
617        Egypt  Rice (Milled Equivalent)  40370
1583     Nigeria  Rice (Milled Equivalent)  26964
1119  Madagascar  Rice (Milled Equivalent)  26118

### Ans: Egypt - 40370, Nigeria - 26964, and Madagascar - 26118